packages/starknet-snap/src/ui/components/ExecuteTxnUI.tsx

```diff
@@ -0,0 +1,168 @@
+import type { SnapComponent } from '@metamask/snaps-sdk/jsx';
+import {
+  Box,
+  Container,
+  Row,
+  Section,
+  Value,
+  Text,
+  Icon,
+} from '@metamask/snaps-sdk/jsx';
+import { formatUnits } from 'ethers/lib/utils';
+
+import type { FeeToken } from '../../types/snapApi';
+import { DEFAULT_DECIMAL_PLACES } from '../../utils/constants';
+import { AddressUI, JsonDataUI, RowUI } from '../fragments';
+import { FeeTokenSelector } from '../fragments/FeeTokenSelector';
+
+/**
+ * The form errors.
+ *
+ * @property to - The error for the receiving address.
+ * @property amount - The error for the amount.
+ * @property fees - The error for the fees.
+ */
+export type ExecuteTxnUIErrors = {
+  to?: string;
+  amount?: string;
+  fees?: string;
+};
+
+export type ExecuteTxnUIProps = {
+  type: string;
+  signer: string;
+  chainId: string;
+  maxFee: string;
+  calls: any[];
+  feeToken: string;
+  includeDeploy: boolean;
+  errors?: ExecuteTxnUIErrors;
+};
+
+/**
+ * A component for executing transactions, providing details and options to configure the transaction.
+ *
+ * @param props - The component props.
+ * @param props.signer - The signer for the transaction.
+ * @param props.chainId - The ID of the chain for the transaction.
+ * @param props.maxFee - The maximum fee allowed for the transaction.
+ * @param props.calls - The calls involved in the transaction.
+ * @param props.feeToken - The token used for fees.
+ * @param props.includeDeploy - Whether to include account deployment in the transaction.
+ * //param props.errors : TODO : add this param
+ * @returns The ExecuteTxnUI component.
+ */
+export const ExecuteTxnUI: SnapComponent<ExecuteTxnUIProps> = ({
+  signer,
+  chainId,
+  maxFee,
+  calls,
+  feeToken,
+  includeDeploy,
+  // errors, // TODO: include this later
+}) => {
+  const tokenTotals: Record<string, { amount: number; decimals: number }> = {};
+
+  // Sum up each call's transfer amount, converting to float
+  calls.forEach((call) => {
+    if (call.isTransfer) {
+      if (!tokenTotals[call.tokenSymbol]) {
+        tokenTotals[call.tokenSymbol] = { amount: 0, decimals: call.decimals };
+      }
+      tokenTotals[call.tokenSymbol].amount += parseFloat(call.amount);
+    }
+  });
+
+  // Convert maxFee (BigNumber) to float and add to feeToken's total if it's present
+  const feeTokenAmount = parseFloat(
+    formatUnits(maxFee, DEFAULT_DECIMAL_PLACES),
```

Comment:
fee can only pay by eth or STRK
it means, the DECIMAL_PLACES should align with the token

Reply:
let use bigint in all calculation